cardano-cli/test/cardano-cli-golden/Test/Golden/CreateTestnetData.hs

```diff
@@ -191,3 +193,43 @@ hprop_golden_create_testnet_data_deleg_non_deleg =
     -- https://github.com/IntersectMBO/cardano-cli/issues/631 witnesses.
 
     H.assertWith (fromIntegral onlyHolderCoin) (\ohc -> ohc > totalSupply `div` 2)
+
+-- Execute this test with:
+-- @cabal test cardano-cli-golden --test-options '-p "/golden create testnet data total supply template/"'@
+--
+-- This test checks that we use the total supply value from the shelley template when --total-supply
+-- is not specified. It was broken in the past (see https://github.com/IntersectMBO/cardano-node/issues/5953).
+hprop_golden_create_testnet_data_total_supply_template :: Property
+hprop_golden_create_testnet_data_total_supply_template =
+  propertyOnce $ moduleWorkspace "tmp" $ \tempDir -> do
+    vanillaShelleyGenesis :: ShelleyGenesis StandardCrypto <-
+      H.readJsonFileOk "test/cardano-cli-golden/files/input/shelley/genesis/genesis.spec.json"
+    let tweakedValue = 3_123_456_000_000
+        tweakedShelleyGenesis = vanillaShelleyGenesis{L.sgMaxLovelaceSupply = tweakedValue}
+        tweakedShelleyGenesisFp = tempDir </> "tweaked-shelley-genesis.json"
+
+    void $ liftIO $ writeFileJSON tweakedShelleyGenesisFp tweakedShelleyGenesis
+
+    let outputDir = tempDir </> "out"
+    void $
+      execCardanoCLI
+        [ "conway"
```

Comment:
We shouldn't hardcore the era here. Use `eraToString`.

Reply:
Actually there is no era witness floating around here. This is the one place where we define the era. (and by the way, `eraToString` is not a thing in CLI nor API, it's a `cardano-testnet` thing)(which maybe we should move to API?).

So I'm merging like this.